libtor/src/lib.rs

```diff
@@ -393,6 +396,40 @@ impl Tor {
     }
 }
 
+/// Generate a hashed password to use HashedControlPassword
+pub fn generate_hashed_password(secret: &str) -> String {
+    // This code is rewrite of
+    // https://gist.github.com/s4w3d0ff/9d65ec5866d78842547183601b2fa4d5
+    // s4w3d0ff and jamesacampbell, Thank you!
+
+    let slen = 8 + (secret.len());
+    let mut salt = Vec::with_capacity(slen);
+    salt.extend_from_slice(&rand::rngs::OsRng.gen::<u64>().to_ne_bytes());
+    salt.extend_from_slice(secret.as_bytes());
```

Comment:
Not sure if I'm reading the python script wrong, but I think they do `rand + 'c' + secret`, but here you never concatenate the "indicator" char

Reply:
I think it not need to concatenate the indicator.
Because in the python script, the salt[8] is not used.
c = ord(salt[8]) means c = 96.
So I used salt as tmp.